src/main/java/uk/gov/justice/hmpps/prison/service/BedAssignmentHistoryService.java

```diff
@@ -78,10 +80,24 @@ public BedAssignmentHistoryPK add(final Long bookingId, final Long livingUnitId,
             .collect(Collectors.toList());
     }
 
-    public List<BedAssignment> getBedAssignmentsHistoryByDate(final LocalDate assignmentDate) {
-        return repository
-            .findBedAssignmentHistoriesByAssignmentDate(assignmentDate)
+    @VerifyAgencyAccess
+    public List<BedAssignment> getBedAssignmentsHistoryByDateForAgency(final String agencyId, final LocalDate assignmentDate) {
+        final var assignments = new ArrayList<>(repository.findBedAssignmentHistoriesByAssignmentDate(assignmentDate));
+
+        final var livingUnitIds = assignments.stream()
+            .map(BedAssignmentHistory::getLivingUnitId)
+            .collect(Collectors.toSet());
+
+        final var agencyLivingUnitIds = livingUnitIds
+            .stream()
+            .map(locationRepository::findOneByLocationId)
```

Comment:
This is going to make 1 call for each location found - is that ok?

Reply:
I thought it was going to be but after testing it... not so much.